Extractor/Pushers/CDFPusher.cs

```diff
@@ -1253,18 +1253,18 @@ private async Task MarkReferencesAsDeleted(IEnumerable<string> externalIds, Canc
         /// </summary>
         private async Task EnsureConfigInit(CancellationToken token)
         {
-            if (!config.DataSetId.HasValue && !string.IsNullOrEmpty(config.DataSetExternalId))
+            if (config.DataSet != null)
```

Comment:
The old if block was checking if dataset was defined by externalid, and retrieving the numeric id if necessary. 

This one is checking the top level object, which is also checked in DataSets.GetId method. Should we rework this section a bit?

Reply:
It's the same, the GetId method checks if ExternalId is defined.